frontend/pages/LabelPage/LabelPage.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useState, useContext, useEffect } from "react";
+import { useQuery } from "react-query";
+import { InjectedRouter, Params } from "react-router/lib/Router";
+
+import PATHS from "router/paths";
+import MainContent from "components/MainContent";
+import SidePanelContent from "components/SidePanelContent";
+import QuerySidePanel from "components/side_panels/QuerySidePanel";
+import { QueryContext } from "context/query";
+import { NotificationContext } from "context/notification";
+import { IApiError } from "interfaces/errors";
+import { ILabel, ILabelFormData } from "interfaces/label";
+import labelsAPI, { ILabelsResponse } from "services/entities/labels";
+import deepDifference from "utilities/deep_difference";
+import LabelForm from "./LabelForm";
+
+const baseClass = "label-page";
+
+interface ILabelPageProps {
+  router: InjectedRouter;
+  params: Params;
+  location: {
+    pathname: string;
+  };
+}
+
+const DEFAULT_CREATE_LABEL_ERRORS = {
+  name: "",
+};
+
+const LabelPage = ({
+  router,
+  params,
+  location,
+}: ILabelPageProps): JSX.Element | null => {
+  const isEditLabel = !location.pathname.includes("new");
+
+  const [selectedLabel, setSelectedLabel] = useState<ILabel>();
+  const [isSidebarOpen, setIsSidebarOpen] = useState<boolean>(true);
+  const [
+    showOpenSchemaActionText,
+    setShowOpenSchemaActionText,
+  ] = useState<boolean>(false);
+  const [labelValidator, setLabelValidator] = useState<{
+    [key: string]: string;
+  }>(DEFAULT_CREATE_LABEL_ERRORS);
+  const { selectedOsqueryTable, setSelectedOsqueryTable } = useContext(
+    QueryContext
+  );
+
+  const { renderFlash } = useContext(NotificationContext);
+
+  const showSidebar = isSidebarOpen;
+
+  const { data: labels, error: labelsError } = useQuery<
+    ILabelsResponse,
+    Error,
+    ILabel[]
+  >(["labels"], () => labelsAPI.loadAll(), {
+    select: (data: ILabelsResponse) => data.labels,
+    onSuccess: (responseLabels: ILabel[]) => {
+      if (params.label_id) {
+        const selectLabel = responseLabels.find(
+          (label) => label.id === parseInt(params.label_id, 10)
+        );
+        setSelectedLabel(selectLabel);
+      }
+    },
+  });
+
+  useEffect(() => {
+    setShowOpenSchemaActionText(!isSidebarOpen);
+  }, [isSidebarOpen]);
```

Comment:
setShowOpenSchemaActionText can be called in the handlers of onCloseSchemaSidebar and onOpenSchemaSidebar. this will allow us to remove this useEffect.